### PR TITLE
Docs for Logs templates and mappings

### PR DESCRIPTION
**Pull request by gr0**
Hi guys, 

This PR introduces templates and mappings dedicated section to Logs docs. It shows how to use templates and what it means to use mappings. I have another docs JIRA on me regarding the fields editor, which will bring some additional information on data structure and SC Logs. I think that when it comes to templates this is enough. 

The page looks as follows:

templates

### CODE REVIEW

File: docs/logs/mappings-templates.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ }
+}'
+```
+
+You need to provide your [write token](/logs/settings/) in the URI of the request and the mappings definition. The **order** property needs to be higher than **20**. The ones below that value are reserved for internal Sematext platform purposes. 
+
+Once you successfully create a new template your index will be rolled over. That means that an internal mechanism will create a new index and will apply the new template for you. 
+
+## The Mappings
+The mappings mechanism in Elasticsearch works by providing the data structure during index creation or by updating the data structure for an already created index. It wouldn't work well in the case of the Sematext Logs platform, because we try to provide you with the best indexing and searching experience possible, and thus we continuously create new indices in the background.
```
Comment (otisg) [inline docs/logs/mappings-templates.md:31]: I think we need to make it clear when to use Templates and when to use Mappings.  I can't tell from these docs so far and I'd be confused.  I'd like to simplify the wording, but I think I would first need to know this basic info.

File: docs/logs/mappings-templates.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+You need to provide your [write token](/logs/settings/) in the URI of the request and the mappings definition. The **order** property needs to be higher than **20**. The ones below that value are reserved for internal Sematext platform purposes. 
+
+Once you successfully create a new template your index will be rolled over. That means that an internal mechanism will create a new index and will apply the new template for you. 
+
+## The Mappings
+The mappings mechanism in Elasticsearch works by providing the data structure during index creation or by updating the data structure for an already created index. It wouldn't work well in the case of the Sematext Logs platform, because we try to provide you with the best indexing and searching experience possible, and thus we continuously create new indices in the background. 
+
+Because of that, the mappings functionality in Sematext Logs works differently. When you send the mappings request it is translated into a template creation request.
```
Comment (otisg) [inline docs/logs/mappings-templates.md:33]: If we translate a mapping to a template, does this mean we could simplify things and tell users to always use the template approach and not even bother with the mapping call?

File: docs/logs/mappings-templates.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        "email_write": {
+          "type": "keyword"
+        }
+      }
+    }
+  } 
+}'
+```
+
+You can find your [API key](/api) in the Settings section of Sematext platform and your Logs write token can be found in the token section of your Logs App.
```
Comment (otisg) [inline docs/logs/mappings-templates.md:51]: You can find your [API key](/api) in the Settings section of Sematext platform and your Logs write token can be found in the token section of your Logs App. 
=>
You can find your [API](/api) key in your [Account settings](https://apps.sematext.com/ui/account/api) ([EU](https://apps.eu.sematext.com/ui/account/api)) and your Logs write token in each Logs App's Settings.

File: docs/logs/mappings-templates.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        }
+      }
+    }
+  } 
+}'
+```
+
+You can find your [API key](/api) in the Settings section of Sematext platform and your Logs write token can be found in the token section of your Logs App. 
+
+In the background, the mappings definition will be changed into a new template. Sematext Logs does that to ensure that your mapping will be moved when we create a new, internal index for your Logs App.
```
Comment (otisg) [inline docs/logs/mappings-templates.md:53]: Implementation detail?

File: mkdocs.yml [hunk 5]
```diff
@@ -42,6 +42,7 @@ pages:
       - Fields: logs/fields.md
       - Field Types: logs/field-types.md
       - Supported Date Formats: logs/supported-date-formats.md
+      - Mappings and Templates: logs/mappings-templates.md
```
Comment (otisg) [inline mkdocs.yml:45]: Elasticsearch terminology
Would this work:
Fields & Types

Ah, funny.  I see we already have a Fields page and Field Types page.  What you wrote in this new mappings-templates.md is very much about fields and field types.  Do we really need a new page?  The Fields page is super basic and empty.  It may make more sense to add this info there instead of having this new page.

**Review by otisg — CHANGES_REQUESTED** [review]

File: docs/logs/mappings-templates.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+title: Mappings and Templates
+description: Field mappings and indices templates supported by Sematext Cloud based SaaS / On-premises logging as a service platform
```
[the remark this answers is not in the document]
Comment (gr0) [reply docs/logs/mappings-templates.md:2]: Done.

File: docs/logs/mappings-templates.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+Every log event that you ship to Sematext Logs has its structure - it is divided into fields. Each field can have a [type](/logs/field-types/), for example [String](/logs/field-types/#string), [Date](/logs/field-types/#date), or [Integer](/logs/field-types/#floatdouble). It can even be an object holding structured data. We do everything we can to ensure that the fields of your logs have the correct types. However, there are situations where you know better what kind of fields you need and which types of data you want to use.
```
[the remark this answers is not in the document]
Comment (gr0) [reply docs/logs/mappings-templates.md]: Done.

File: docs/logs/mappings-templates.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
+title: Mappings and Templates
+description: Field mappings and indices templates supported by Sematext Cloud based SaaS / On-premises logging as a service platform
+
+Every log event that you ship to Sematext Logs has its structure - it is divided into fields. Each field can have a [type](/logs/field-types/), for example [String](/logs/field-types/#string), [Date](/logs/field-types/#date), or [Integer](/logs/field-types/#floatdouble). It can even be an object holding structured data. We do everything we can to ensure that the fields of your logs have the correct types. However, there are situations where you know better what kind of fields you need and which types of data you want to use. 
+
+In such cases, Sematext Logs provide mechanism for handling the data structure - the templates and the mappings. The mentioned functionality comes from Elasticsearch and its compatible with its API, however, because of the nature of the Sematext platform it works differently.
```
[the remark this answers is not in the document]
Comment (gr0) [reply docs/logs/mappings-templates.md]: Done and will adjust the mentioned section.

File: docs/logs/mappings-templates.md (hunk 4 above)
Comment (gr0) [reply docs/logs/mappings-templates.md:53]: Adjusted and removed the information about the implementation. Not needed.

File: docs/logs/mappings-templates.md (hunk 3 above)
Comment (gr0) [reply docs/logs/mappings-templates.md]: Adjusted.

File: docs/logs/mappings-templates.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+## The Templates
+The templates mechanism works by providing guidance for Elasticsearch on the structure of the data. Every time an index is created all the defined templates are used to pre-create the fields with their types in such index. It is no different in Sematext Logs. By defining a template you provide information on what fields and types you expect in your data.
```
[the remark this answers is not in the document]
Comment (gr0) [reply docs/logs/mappings-templates.md]: Done.

File: mkdocs.yml (hunk 5 above)
Comment (gr0) [reply mkdocs.yml:45]: I see it that way - the `Fields` page is about the fields themselves and don't touch the types. The `Types` page describe the types that are possible to be used. I would extend them with Field Editor information, because once it starts working again it should be the way people should modify structure of the data inside their Logs Apps. 

The `Templates and Mappings` are something that is available in addition, for more "power users" let's say. I think it should be a separate section, similar to Elasticsearch documentation is built. I will add an information in those two pages about the `Templates and Mappings` though, to have a link there.

File: docs/logs/mappings-templates.md (hunk 2 above)
Comment (gr0) [reply docs/logs/mappings-templates.md]: Added a sentence like this `Note that the mappings API support is provided by Sematext Logs for compatibility purposes and when changing the structure of your data you should use the templates functionality`.

File: docs/logs/mappings-templates.md (hunk 1 above)
Comment (gr0) [reply docs/logs/mappings-templates.md:33]: I changed the description slightly and added a "note" that the support for mappings API is there mainly for compatibility purpose. Let me know if it makes more sense now.

File: docs/logs/mappings-templates.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+In such cases, Sematext Logs provide mechanism for handling the data structure - the templates and the mappings. The mentioned functionality comes from Elasticsearch and its compatible with its API, however, because of the nature of the Sematext platform it works differently.
+
+## The Templates
+The templates mechanism works by providing guidance for Elasticsearch on the structure of the data. Every time an index is created all the defined templates are used to pre-create the fields with their types in such index. It is no different in Sematext Logs. By defining a template you provide information on what fields and types you expect in your data. 
+
+You can easily create a new template by running the following request:
```
[the remark this answers is not in the document]
Comment (gr0) [reply docs/logs/mappings-templates.md:11]: Added, have a look and let me know if I need to add more information there.

File: docs/logs/mappings-templates.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+``` code
+curl -XPUT 'https://logsene-receiver.sematext.com/_template/YOUR_WRITE_TOKEN_template_name' -d '{
```
[the remark this answers is not in the document]
Comment (adnanrahic) [reply docs/logs/mappings-templates.md:14]: I agree this is confusing. We should just have `LOGS_TOKEN`.

File: docs/logs/mappings-templates.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+ }
+}'
+```
+
+---
+**Note:**
+For EU location use **logsene-receiver.eu.sematext.com** instead of **logsene-receiver.sematext.com**.
+---
+
+You need to provide your [write token](/logs/settings/) in the URI of the request, the **order**, and the mappings definition. The **order** property needs to be higher than **20**. The ones below that value are reserved for internal Sematext platform purposes and result in error when being used.
```
Comment (adnanrahic) [inline docs/logs/mappings-templates.md:31]:
```suggestion
You need to provide your [`LOGS_TOKEN`](/logs/settings/) in the URI of the request, the **order**, and the mappings definition. The **order** property needs to be higher than **20**. The ones below that value are reserved for internal Sematext platform purposes and result in error when being used.
```

I'd rather not use write token.

File: docs/logs/mappings-templates.md [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+
+### Template Structure
+The main section of each template definition is the **mappings** section which contains information about the fields and their types. Your templates may include one or more fields fields that are not considered special (more information in the [fields](/logs/fields)). For example:
```
Comment (adnanrahic) [inline docs/logs/mappings-templates.md:34]:
```suggestion
The main section of each template definition is the **mappings** section which contains information about the fields and their types. Your templates may include one or more fields that are not considered special. More information about this in the [fields section](/logs/fields). For example:
```

File: docs/logs/mappings-templates.md [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+
+``` code
+curl -XPUT 'https://logsene-receiver.sematext.com/_template/YOUR_WRITE_TOKEN_mytemplate' -d '{
```
Comment (adnanrahic) [inline docs/logs/mappings-templates.md:37]:
```suggestion
curl -XPUT 'https://logsene-receiver.sematext.com/_template/LOGS_TOKEN_mytemplate' -d '{
```

File: docs/logs/mappings-templates.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+     },
+     "user_description": {
+       "type": "text"
+     }
+   }
+ }
+}'
+```
+
+The above template contains the **mappings** object. Inside it we fine the **properties** object which is responsible for defining the fields and their types. Each field is a JSON object that contains a name, like **user_name** in the above example and the type defined by using the **type** keyword.
```
Comment (adnanrahic) [inline docs/logs/mappings-templates.md:58]:
```suggestion
The template above contains the **mappings** object. Inside it we find the **properties** object which is responsible for defining the fields and their types. Each field is a JSON object that contains a name, like **user_name**. The example above we see the type defined by using the **type** keyword. 
```

File: docs/logs/mappings-templates.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+
+---
+**NOTE:**
+The mappings API support is provided by Sematext Logs for compatibility purposes and when changing the structure of your data you should use the [templates](/logs/mappings-templates/#the-templates) functionality.
+---
+
+For example this command results in a mappings request made to Sematext Cloud:
+
+``` code
+curl -H 'Authorization: apiKey YOUR_API_KEY' -XPUT 'https://logsene-receiver.sematext.com/YOUR_WRITE_TOKEN/_mapping' -d '{
```
Comment (adnanrahic) [inline docs/logs/mappings-templates.md:89]:
```suggestion
curl -H 'Authorization: apiKey YOUR_API_KEY' -XPUT 'https://logsene-receiver.sematext.com/LOGS_TOKEN/_mapping' -d '{
```

File: docs/logs/mappings-templates.md [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+        "email_write": {
+          "type": "keyword"
+        }
+      }
+    }
+  } 
+}'
+```
+
+You can find your [API key](/api) in your [Account settings](https://apps.sematext.com/ui/account/api) ([EU](https://apps.eu.sematext.com/ui/account/api)) and your Logs write token in each Logs App's Settings.
```
Comment (adnanrahic) [inline docs/logs/mappings-templates.md:102]:
```suggestion
You can find your [API key](/api) in your [Account settings](https://apps.sematext.com/ui/account/api) ([EU](https://apps.eu.sematext.com/ui/account/api)) and your `LOGS_TOKEN` in each Logs App's Settings.
```

**Review by adnanrahic — CHANGES_REQUESTED** [review]
A few edits.

File: docs/logs/mappings-templates.md (hunk 17 above)
Comment (gr0) [reply docs/logs/mappings-templates.md]: Done.

File: docs/logs/mappings-templates.md (hunk 16 above)
Comment (gr0) [reply docs/logs/mappings-templates.md]: Done.

File: docs/logs/mappings-templates.md (hunk 15 above)
Comment (gr0) [reply docs/logs/mappings-templates.md]: Done.

File: docs/logs/mappings-templates.md (hunk 14 above)
Comment (gr0) [reply docs/logs/mappings-templates.md]: Done.

File: docs/logs/mappings-templates.md (hunk 12 above)
Comment (gr0) [reply docs/logs/mappings-templates.md]: Agreed, changed.

File: docs/logs/mappings-templates.md (hunk 13 above)
Comment (gr0) [reply docs/logs/mappings-templates.md]: Done.

File: mkdocs.yml (hunk 5 above)
Comment (otisg) [reply mkdocs.yml:45]: > The Types page describe the types that are possible to be used. I would extend them with Field Editor information, because once it starts working again it should be the way people should modify structure of the data inside their Logs Apps

Yeah, but FE is not only about changing field types.  It's also about e.g. removing fields.
And if we look at https://sematext.com/docs/logs/fields/ today what do we see? 2 sentences referencing Common Schema, a list of fields, and a screenshot.  If we look at https://sematext.com/docs/tags/common-schema/#logs-tags we see those same fields are already listed there.  Should we even have this Fields page as such then?  IMHO we should:
# Replace the content of Fields page with a paragraph about log events having structure and fields -- basically the first paragraph from this PR, followed by info about Field Editor.
# From this page we'd have references to Field Types page and we would also have references to Mappings & Templates.

File: docs/logs/mappings-templates.md [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+Logs are stored in an index in Sematext. Each Logs App has its own index and its own, independent log event structure. A template provides information about the structure - fields and their types - of log events in a given App and its underlying index. Every time an index is created all the defined templates are used to pre-create the fields with their types in such index. It is no different in Sematext Logs. By defining a template you provide information on what fields and types you expect in your data. 
+
+You can easily create a new template by running the following request:
+
+``` code
+curl -XPUT 'https://logsene-receiver.sematext.com/_template/LOGS_TOKEN_template_name' -d '{
+ "order": 31,
+ "mappings": {
+   "properties": {
+     "host_name": {
```
Comment (otisg) [inline docs/logs/mappings-templates.md:18]: Not os.host?

File: docs/logs/mappings-templates.md [hunk 19]
```diff
@@ -0,0 +1,102 @@
+title: Mappings and Templates
+description: Field mappings and index template supported by Sematext Cloud
+
+Every log event that you ship to Sematext Logs has its structure - it is divided into fields. Each field can have a [type](/logs/field-types/), for example [string](/logs/field-types/#string), [date](/logs/field-types/#date), or [integer](/logs/field-types/#integerlong). It can even be an object holding structured data. We do everything we can to ensure that your log event field types are inferred correctly. However, you may also want to set field types explicitly. 
+
+To do that you can create templates and configure mapping via Sematext's [Elasticsearch-compatible API](/logs/index-events-via-elasticsearch-api/) as shown below.
+
+## The Templates
```
Comment (otisg) [inline docs/logs/mappings-templates.md:8]: The Templates => Index Templates

File: docs/logs/mappings-templates.md [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+ * object
+
+---
+**NOTE:**
+Before adding new fields to your templates ensure that they are not present in the [special fields](/logs/fields) or in the [common schema](/tags/common-schema/).
+---
+
+Once you successfully create a new template your index will be rolled over. That means that an internal mechanism will create a new index and will apply the new template for you. 
+
+## The Mappings
```
Comment (otisg) [inline docs/logs/mappings-templates.md:78]: The Mappings => Index Mappings

**Review by otisg — CHANGES_REQUESTED** [review]

File: docs/logs/mappings-templates.md (hunk 18 above)
Comment (gr0) [reply docs/logs/mappings-templates.md:18]: Badly chosen name of the field. I would avoid any fields from the common schema. Changed.

File: docs/logs/mappings-templates.md (hunk 19 above)
Comment (gr0) [reply docs/logs/mappings-templates.md:8]: Done.

File: docs/logs/mappings-templates.md (hunk 20 above)
Comment (gr0) [reply docs/logs/mappings-templates.md]: Done.

File: mkdocs.yml (hunk 5 above)
Comment (gr0) [reply mkdocs.yml:45]: OK, got it. I just changed it @otisg, have a look now.

Comment (otisg) [reply mkdocs.yml:45]: I think this looks about right.  And I assume FE docs you plan on adding at a later time, si?

Comment (gr0) [reply mkdocs.yml:45]: Yes, FE docs are in another JIRA I had for this sprint, but I will not make it. First - I will make it "great again" and then I'll describe how awesome it is ;)

**Comment by gr0** [comment]
Thanks for the review guys, as there are no further comments, I'll merge that.
